x/evm/types/params.go

```diff
@@ -46,7 +46,7 @@ func DefaultParams() Params {
 		EvmDenom:     DefaultEVMDenom,
 		EnableCreate: true,
 		EnableCall:   true,
-		ExtraEIPs:    []int64{2929, 2200, 1884, 1344},
+		ExtraEIPs:    []int64{1344, 1884, 2200, 2929},
```

Comment:
are these supposed to always be in asc order? If so we could add an additional validation on the params

Reply:
https://github.com/ethereum/go-ethereum/blob/f0b1bddac41a31ff5b86c3d934acd19016aeb1d1/core/vm/jump_table.go#L85

I think it should follow the hard fork order, [Istanbul](https://github.com/ethereum/go-ethereum/blob/f0b1bddac41a31ff5b86c3d934acd19016aeb1d1/core/vm/jump_table.go#L85) do 1344,1884,2200, [Berlin](https://github.com/ethereum/go-ethereum/blob/f0b1bddac41a31ff5b86c3d934acd19016aeb1d1/core/vm/jump_table.go#L77) do 2929.

But my other question is since we already enabled Berlin hardfork in chain config, do we have to specify extra eips here?

Reply:
> But my other question is since we already enabled Berlin hardfork in chain config, do we have to specify the extra eips here?

I think the VM config relies on the extra EIPs for the gas values? not sure what happens if you don't define them and enable the fork on chain config

Reply:
https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L80
Judge from the code, I think we don't need the extra eips.

Reply:
> https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L80
> Judge from the code, I think we don't need the extra eips, I'll have a test.

seems to be the case, after remove the extra eips, the gas number not changed.

Reply:
yeah, it seems that they are not necessary. I would maybe remove them from the default parameters but still define them as a var:

```go
var AvailableExtraEIPs = []int64{1344, 1884, 2200, 2929}
```

Reply:
done